itests/standalone/basic/src/test/java/org/wildfly/camel/test/common/security/SecurityUtils.java

```diff
@@ -0,0 +1,170 @@
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* #L%
+*/
+package org.wildfly.camel.test.common.security;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSession;
+
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wildfly.camel.test.common.utils.FileUtils;
+import org.wildfly.camel.test.common.utils.WildFlyCli;
+import org.wildfly.camel.test.cxf.ws.secure.subA.Application;
+import org.wildfly.camel.test.cxf.ws.secure.subA.CxfWsRouteBuilder;
+import org.wildfly.camel.test.cxf.ws.secure.subA.GreetingService;
+import org.wildfly.camel.test.cxf.ws.secure.subA.GreetingsProcessor;
+
+public class SecurityUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(SecurityUtils.class);
+
+    private static final String APPLICATION_KEYSTORE_PASSWORD = "password";
+
+    private static final String CLIENT_CERT_KEYSTORE_PASSWORD = "123456";
+    private static final String CLIENT_CRT = "client.crt";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final String CLIENT_TRUSTSTORE = "client.truststore";
+    private static final String SERVER_ALIAS = "server";
+    private static final String SERVER_CRT = "server.crt";
+    private static final String SERVER_KEYSTORE = "server.keystore";
+    private static final String SERVER_TRUSTSTORE = "server.truststore";
+    private static final String UNTRUSTED_CRT = "untrusted.crt";
+    private static final String UNTRUSTED_KEYSTORE = "untrusted.keystore";
+
+    private static final String JBOSS_WEB_XML_TEMPLATE = "<jboss-web><security-domain>%s</security-domain></jboss-web>";
+    private static final String WEB_XML_TEMPLATE = "<web-app>" //
+            + "<security-constraint>" //
+            + "<display-name>SecurityConstraint</display-name>" //
+            + "<web-resource-collection>" //
+            + "<web-resource-name>All Resources</web-resource-name>" //
+            + "<url-pattern>///*</url-pattern>" //
```

Comment:
Possibly a silly question but I want to make sure we have this covered....

There could potentially be multiple CXF endpoints configured within the deployment. This solution can handle applying security to multiple paths, right?

E.g if I were to configure CXF endpoint 'A' to /foo/bar and endpoint 'B' to /foo/bar/cheese, can I apply individual security constraints to each individual path?

Reply:
`///` in `<url-pattern>///*</url-pattern>` is actually a marker for the subsystem to know that the constraint is to be applied to a CXF endpoint. I see this myself as a semilegal shorthand. That said I still consider the question open where/how the security aspects of a CXF endpoint should be defined.

And back to your question: yes, /foo/bar and /foo/bar/cheese have 100% independent security constraints. I should add a test.

Reply:
I have added a test that /foo/bar and /foo/bar/cheese can have independent security constraints.

We should decide where/how the security aspects of a CXF endpoint should be defined. The trick with `///` prefix in `web.xml` works, but I wonder if we can come up with a similarly easy but more Java EE conform solution.

Alternatively, we could have separate web.xml files per endpoint: there would be `web-inf/foo/bar/web.xml` in the WAR for the /foo/bar endpoint. Cumbersome, but IMO less illegal.

You mentioned in the past that the security aspects could be configured via URI params of the camel endpoint. I still cannot imagine how that could work, TBH. I'd appreciate some semi-formal example if you still prefer that.

Reply:
Maybe endpoint URI params aren't appropriate here.

Would it be simpler to have a custom descriptor file for our security configuration?

Reply:
> Would it be simpler to have a custom descriptor file for our security configuration?

That would solve the problem, would be clean in sense not misusing EE config files for something they were not designed for. However, that's more coding for us. Are you in favor of this solution?